### PR TITLE
Feature/domain

### PR DESCRIPTION
**Pull request by sarisiza**
Created the Domain class for the Drinks

### CODE REVIEW

File: app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        category = this?.strCategory,
+        drinkName = this?.strDrink,
+        drinkThumb = this?.strDrinkThumb+"/preview",
+        glass = this?.strGlass,
+        ingredients = ingredientsList,
+        measures = measuresList,
+        instructions = this?.strInstructions
+    )
+
+}
```
Comment (carlosCabrera2) [inline app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt:72]: Very easy to read and nicely organized.

File: app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    val category: String?,
+    val drinkName: String?,
+    val drinkThumb: String?,
+    val glass: String?,
+    val ingredients: List<String?>?,
+    val instructions: String?,
+    val measures: List<String?>?
+)
+
+fun CocktailsResponse?.mapToDrinkDomain(): Drink{
```
Comment (carlosCabrera2) [inline app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt]: in the model it is not called "mapToDrinkDomain". i believe that we need to change it . Right?

**Review by carlosCabrera2 — DISMISSED** [review]
it looks good to me

File: app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt (hunk 2 above)
Comment (sarisiza) [reply app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt]: Ok, I'll change it

**Review by chr1s4ng3l — DISMISSED** [review]

**Review by carlosCabrera2 — DISMISSED** [review]

File: app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            this?.strMeasure8,
+            this?.strMeasure9,
+            this?.strMeasure10,
+            this?.strMeasure11,
+            this?.strMeasure12,
+            this?.strMeasure13,
+            this?.strMeasure14,
+            this?.strMeasure15
+        )
+    measuresList.forEach{
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt]: what about using a filter function

File: app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        if(it == null)
+            measuresList.remove(it)
+    }
+
+    return Drink(
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt:60]: this code might be reduced by using scope functions "run", "let", "also", "apply" or. "with"

File: app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt (hunk 3 above)
Comment (inaki92) [reply app/src/main/java/com/example/thecocktailsapp/model/domain/Drink.kt]: list.filter {  it != null }

**Review by inaki92 — DISMISSED** [review]
take a look at the comments

**Comment by sarisiza** [comment]
Tried to push changes on Domain, and previous commits got pushed. Recreating the branch and pushing again

**Pull request closed without merging**